pkg/provider/alibaba/kms.go

```diff
@@ -302,19 +302,6 @@ func (kms *KeyManagementService) Validate() (esv1beta1.ValidationResult, error)
 	return esv1beta1.ValidationResultReady, nil
 }
 
-func (kms *KeyManagementService) ValidateStore(store esv1beta1.GenericStore) error {
-	storeSpec := store.GetSpec()
-	alibabaSpec := storeSpec.Provider.Alibaba
-
-	regionID := alibabaSpec.RegionID
-
-	if regionID == "" {
-		return fmt.Errorf("missing alibaba region")
-	}
-
-	return kms.validateStoreAuth(store)
-}
-
 func (kms *KeyManagementService) validateStoreAuth(store esv1beta1.GenericStore) error {
```

Comment:
When moving out the `ValidateStore()` (which is a good idea!) please also move the `validateStoreAuth` to keep it coherent